### PR TITLE
adding block time research

### PR DESCRIPTION
**Pull request by sbvegan**
**Description**

Adding Sunnyside's block time research into the docs site: https://docs.google.com/document/d/13dc1x9Vwc4BOq2M1Z-VNKb7w1xFHEEeeKyZ2nX8YnF4/edit?usp=sharing

### CODE REVIEW

File: pages/stack/research/block-time-research.mdx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+This [GitHub repository](https://github.com/testinprod-io/block-building-stats/tree/main) contains the test methods, data sets, client versions, and raw data.
+
+The following benchmarks are available in this [notebook](https://github.com/testinprod-io/block-building-stats/blob/main/stats.ipynb).
+
+## Benchmarks
+
+![Figure 1: op-geth / archive node / block 5492540 \~ 9816497
+.](/img/op-stack/protocol/block-time-research-figure-1.png)
+
+**Figure 1**: op-geth / archive node / block 5492540 \~ 9816497
```
Comment (bradleycamacho) [inline pages/stack/research/block-time-research.mdx:25]:
```suggestion
| ![Figure 1: op-geth / archive node / block 5492540 \~ 9816497.](/img/op-stack/protocol/block-time-research-figure-1.png)| 
|:--:| 
| **Figure 1**: op-geth / archive node / block 5492540 \~ 9816497 |
```

Comment (bradleycamacho) [reply pages/stack/research/block-time-research.mdx:25]: Screenshot 2024-10-28 at 1 21 50 PM

This was really hard to read originally. Figure 1 implement this really hacky image caption format, but we should create our own component soon.

Comment (bradleycamacho) [reply pages/stack/research/block-time-research.mdx:25]: Suggesting these for the rest, but happy to chat about this if needed

File: pages/stack/research/block-time-research.mdx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+![Figure 2: op-geth / full node / block 5492540 \~ 9816497.](/img/op-stack/protocol/block-time-research-figure-2.png)
+
+**Figure 2**: op-geth / full node / block 5492540 \~ 9816497
```
Comment (bradleycamacho) [inline pages/stack/research/block-time-research.mdx:29]:
```suggestion
| ![Figure 2: op-geth / full node / block 5492540 \~ 9816497.](/img/op-stack/protocol/block-time-research-figure-2.png) | 
|:--:| 
| **Figure 2**: op-geth / full node / block 5492540 \~ 9816497 |
```

File: pages/stack/research/block-time-research.mdx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+Figures 1 and 2 show the Base nodes' block-building time distribution with op-geth archive node & full node from block 5492540 to 9816497. We can see that the average block building time takes 0.58 and 0.36 seconds each for blocks that spent 25M \~ 30M gas, which is less than one second.
+
+![Figure 3: op-reth / archive node / block 5492540 \~ 9816497.](/img/op-stack/protocol/block-time-research-figure-3.png)
+
+**Figure 3**: op-reth / archive node / block 5492540 \~ 9816497
```
Comment (bradleycamacho) [inline pages/stack/research/block-time-research.mdx:35]:
```suggestion
| ![Figure 3: op-reth / archive node / block 5492540 \~ 9816497.](/img/op-stack/protocol/block-time-research-figure-3.png) | 
|:--:| 
| **Figure 3**: op-reth / archive node / block 5492540 \~ 9816497 |
```

File: pages/stack/research/block-time-research.mdx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+Figure 3 shows the Base nodes' block-building time distribution using the op-reth archive node from block 5492540 to 9816497. Compared to op-geth's archive node, we can see that op-reth shows a better performance in all ranges.
+
+![Figure 4: op-geth / archive node / block 13686867 \~ 15074141.](/img/op-stack/protocol/block-time-research-figure-4.png)
+
+**Figure 4:** op-geth / archive node / block 13686867 \~ 15074141
```
Comment (bradleycamacho) [inline pages/stack/research/block-time-research.mdx:41]:
```suggestion
| ![Figure 4: op-geth / archive node / block 13686867 \~ 15074141.](/img/op-stack/protocol/block-time-research-figure-4.png) | 
|:--:| 
| **Figure 4:** op-geth / archive node / block 13686867 \~ 15074141 |
```

File: pages/stack/research/block-time-research.mdx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+![Figure 5: op-geth / full node / block 14567037 \~ 15074141.](/img/op-stack/protocol/block-time-research-figure-5.png)
+
+**Figure 5:** op-geth / full node / block 14567037 \~ 15074141
```
Comment (bradleycamacho) [inline pages/stack/research/block-time-research.mdx:45]:
```suggestion
| ![Figure 5: op-geth / full node / block 14567037 \~ 15074141.](/img/op-stack/protocol/block-time-research-figure-5.png) | 
|:--:| 
| **Figure 5:** op-geth / full node / block 14567037 \~ 15074141 |
```

File: pages/stack/research/block-time-research.mdx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+Throughout the research, we found that the node meaningfully takes longer to build a block as the chain stores more states and transactions to access more historical data. Therefore, we benchmarked the latest blocks in Figures 4 and 5. On average, both the full node and archive node could build a congested block on time. It is worth noting that the average block-building time of high gas spending range is similar to the older blocks, but the average block-building time is higher on the newer blocks.
+
+![Figure 6: op-geth / archive node / block 13686867 \~ 15074141 / histogram of 25m\~30m gas range.](/img/op-stack/protocol/block-time-research-figure-6.png)
+
+**Figure 6**: op-geth / archive node / block 13686867 \~ 15074141 / histogram of 25m\~30m gas range
```
Comment (bradleycamacho) [inline pages/stack/research/block-time-research.mdx:51]:
```suggestion
| ![Figure 6: op-geth / archive node / block 13686867 \~ 15074141 / histogram of 25m\~30m gas range.](/img/op-stack/protocol/block-time-research-figure-6.png) | 
|:--:| 
| **Figure 6**: op-geth / archive node / block 13686867 \~ 15074141 / histogram of 25m\~30m gas range |
```

**Review by bradleycamacho — CHANGES_REQUESTED** [review]
The images made this really hard to read, so I recommended a fix for them. The rest of the content looks fine for the audience who'd read this. GJ! Happy to approve one I take a look at the new preview

File: pages/stack/operators.mdx
Comment (bradleycamacho) [inline pages/stack/operators.mdx:1]: Also, what's going on with all these deleted files in the PR?

Comment (sbvegan) [reply pages/stack/operators.mdx:1]: Oh yeah, I forgot I had this clean up in this PR. These are outdated breadcrumb autogenerated files that got missed when the OP Stack section was refactored.

**Pull request merged**